lib/spack/spack/detection/common.py

```diff
@@ -197,10 +200,41 @@ def update_configuration(detected_packages, scope=None, buildable=True):
     return all_new_specs
 
 
+def find_windows_compiler_root_paths():
+    """Helper for Windows compiler installation root discovery
+
+    At the moment simply returns location of VS install paths from VSWhere
+    But should be extended to include more information as relevant"""
+    return list(winOs.WindowsOs.vs_install_paths)
+
+
+def find_windows_compiler_cmake_paths():
```

Comment:
I think it would be useful to consolidate these `find_windows_..._paths` functions into an object to make the module more readable

Reply:
Great idea. On it.